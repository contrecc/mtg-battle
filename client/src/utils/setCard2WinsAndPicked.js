export default function setCard2WinsAndPicked() {
  return function(state, props) {
    return {
      winner: true,
      winStreak: state.winStreak + 1,
      winningCard: 'card2',
      roundCompleted: true
    };
  };
}
