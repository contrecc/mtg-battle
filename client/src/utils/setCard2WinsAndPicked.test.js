import setCard2WinsAndPicked from './setCard2WinsAndPicked';

describe('setCard2WinsAndPicked', () => {
  it('should return a function', () => {
    expect(setCard2WinsAndPicked()).toMatchSnapshot();
  });

  it('should return a function that then updates the state', () => {
    const returnedFunction = setCard2WinsAndPicked();
    expect(returnedFunction({ winStreak: 1 }, {})).toMatchSnapshot();
  });
});
