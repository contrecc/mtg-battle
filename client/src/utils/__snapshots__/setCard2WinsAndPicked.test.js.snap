// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setCard2WinsAndPicked should return a function 1`] = `[Function]`;

exports[`setCard2WinsAndPicked should return a function that then updates the state 1`] = `
Object {
  "roundCompleted": true,
  "winStreak": 2,
  "winner": true,
  "winningCard": "card2",
}
`;
